src/Engine/EmitMSIL/Replication.cs

```diff
@@ -23,7 +113,13 @@ public class Replication
         /// <returns></returns>
         public static IList ReplicationLogic(IEnumerable<MethodBase> mInfos, IList args, string[][] replicationAttrs)
         {
-            var reducedArgs = ReduceArgs(args);
+            // Static instance of MSILRuntimeCore
+            // TODO_MSIL: FIgure out how and when to set this runtimeCore (CodeGen should have it too)
```

Comment:
what do you imagine the codegen will use the runtime core for?

Reply:
CodeGEn can add more type information  (class, primitives, methods etc) to the RuntimeCore so that it can be later used during runtime.
The old VM populates its Type information during EMITImportStatement and EMitClassDefinition stages of the compiler.